src/main/java/seedu/address/logic/commands/OpenLinkCommand.java

```diff
@@ -0,0 +1,150 @@
+     */
+    public OpenLinkCommand(ModuleCode moduleCode, List<String> linkAliases) {
+        requireAllNonNull(moduleCode, linkAliases);
+        this.moduleCode = moduleCode;
+        this.linkAliases = linkAliases;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Module moduleToOpenLinks = null;
+        try {
+            moduleToOpenLinks =
+                    model.getModuleUsingModuleCode(moduleCode, true);
+        } catch (ModuleNotFoundException e) {
+            throw new CommandException(String.format(MESSAGE_NO_MODULE_IN_FILTERED_LIST,
+                    moduleCode.getModuleCodeAsUpperCaseString()));
+        }
+        assert moduleToOpenLinks != null;
+
+        String linksOpened = openLinksFromModule(moduleToOpenLinks, linkAliases);
+        return new CommandResult(String.format(MESSAGE_OPEN_LINK_SUCCESS,
+                moduleCode.getModuleCodeAsUpperCaseString(), linksOpened));
+    }
+
+    /**
+     * Iterates through the links from {@code moduleToOpenLinks} and opens it in the user's default browser
+     * if its link alias is found in {@code linksToOpen}.
+     * Partial opening of links is supported, whereby links will be sequentially opened till the end of
+     * the collection {@code linksToOpen} or till a non-matching link alias between the collection and
+     * the links from the specified module is encountered.
+     */
+    private static String openLinksFromModule(Module moduleToOpenLinks, List<String> linkAliasesToOpen)
+            throws CommandException {
+        assert moduleToOpenLinks != null;
+        Set<Link> moduleLinksCopy = moduleToOpenLinks.copyLinks();
+        StringBuilder linksOpened = new StringBuilder();
+        for (String linkAlias : linkAliasesToOpen) {
+            Link linkToOpen = moduleLinksCopy.stream()
+                    .filter(link -> link.hasLinkAlias(linkAlias))
+                    .findFirst()
+                    .orElseThrow(() -> new CommandException(String.format(MESSAGE_MISSING_LINK_ALIAS,
+                            linkAlias, moduleToOpenLinks.getModuleCode().getModuleCodeAsUpperCaseString())));
+            try {
+                launchLinkFromPlannit(linkToOpen.linkUrl);
+                linksOpened.append(String.format("[%s]\n", linkToOpen.linkUrl));
+            } catch (IOException | SecurityException ex) {
+                throw new CommandException(MESSAGE_LINK_LAUNCH_FAILURE);
+            }
+        }
+        return linksOpened.toString();
+    }
+
+    //@@author shwene-reused
+    //Reused from https://stackoverflow.com/questions/5226212/how-to-open-the-default-webbrowser-using-java
+    //with slight modification
+    private static void launchLinkFromPlannit(String linkUrl) throws IOException, SecurityException {
+        boolean isLinkUrlPaddedWithHttps = linkUrl.startsWith(LINK_HEADER_PROTOCOL_HTTPS);
```

Comment:
```suggestion
        assert linkUrl != null or requireNonNull(linkUrl)
        boolean isLinkUrlPaddedWithHttps = linkUrl.startsWith(LINK_HEADER_PROTOCOL_HTTPS);
```
Should we use an assertion here for defensive programming?